yt_dlp/extractor/nhk.py

```diff
@@ -334,3 +335,140 @@ def _real_extract(self, url):
+        'skip_download': True,
+    }, {
+        'url': 'https://www.nhk.or.jp/radio/player/ondemand.html?p=F300_06_3738470',  # one with letters in the id
+        '_type': 'video',
+        'skip': 'Expires on 2024-03-31',
+        'info_dict': {
+            'id': 'F300_06_3738470',
+            'ext': 'm4a',
+            'title': '有島武郎「一房のぶどう」',
+            'description': '朗読：川野一宇（ラジオ深夜便アンカー）\r\n\r\n（2016年12月8日放送「ラジオ深夜便『アンカー朗読シリーズ』」より）',
+            'channel': 'NHKラジオ第1、NHK-FM',
+            'timestamp': 1635757200,
+            'thumbnail': 'https://www.nhk.or.jp/radioondemand/json/F300/img/corner/box_109_thumbnail.jpg',
+            'release_date': '20161207',
+            'series': 'らじる文庫 by ラジオ深夜便 ',
+            'release_timestamp': 1481126700,
+            'upload_date': '20211101',
+        }
+    }
+    ]
+# https://www.nhk.or.jp/radionews/ known not working - doesnt use same api
+# https://www.nhk.or.jp/s-media/news/podcast/list/v1/all.xml podcast feed if you need it
+
+    def _decode_time(self, time):
+        time_format = '%Y-%m-%dT%H:%M:%S%z'
+        try:
+            return datetime.strptime(time, time_format).timestamp()
+        except Exception:
+            # oh well, didnt work
+            pass
+
+    def _extract_episode_info(self, headline, programme_id):
+        file = headline['file_list'][0]
+        # this will break if there's an episode with multiple files, but i don't think that's ever actually the case
+
+        info = {}
+
+        url = file['file_name']
+        info['id'] = programme_id + '_' + headline['headline_id']
+        info['formats'] = self._extract_m3u8_formats(url, info['id'])
+        info['title'] = file.get('file_title')
+        info['description'] = file.get('file_title_sub')
+        # info['webpage_url'] = file.get('share_url')
+        info['thumbnail'] = headline.get('headline_image')
+        info['container'] = 'm4a_dash'  # force fixup so seeking works
+        info['was_live'] = True  # it's radio catch-up
+
+        aa_vinfo4 = file.get('aa_vinfo4')
+        # have to use whatever this is to get an actual air date
+        # there is an onair_date var, but it's natural language that doesnt have half of what we need anyway
+        info['release_timestamp'] = self._decode_time(aa_vinfo4.split('_')[0])
+        # open_time is when it was put on vod
+        info['timestamp'] = self._decode_time(file.get('open_time'))
+
+        return info
+
+    def _real_extract(self, url):
+        video_id, site_id, corner_id, headline_id = self._match_valid_url(url).groups()
+
+        programme_id = f'{site_id}_{corner_id}'
+        json_url = f'https://www.nhk.or.jp/radioondemand/json/{site_id}/bangumi_{programme_id}.json'
+        meta = self._download_json(json_url, programme_id)['main']
+
+        series = meta.get('program_name')
+        channel = meta.get('media_name')
+        thumbnail = meta.get('thumbnail_c') or meta.get('thumbnail_p')
```

Comment:
A dict will be easier to pass around. We can pass it to `_extract_episode_info` and let that method handle the info_dict update for single episodes and playlist entries, and also pass it to the playlist result
```suggestion
        series_meta = traverse_obj(meta, {
            'title': 'program_name',
            'channel': 'media_name',
            'thumbnail': (('thumbnail_c', 'thumbnail_p'), {url_or_none}),
        }, get_all=False)
```